### PR TITLE
Update to gstreamer-rs 0.14

### PR DESCRIPTION
**Pull request by ceyusa**

### CODE REVIEW

File: backends/gstreamer/audio_decoder.rs [hunk 1]
```diff
@@ -220,7 +218,7 @@ impl AudioDecoder for GStreamerAudioDecoder {
                                     })?;
 
                                     for position in positions.iter() {
-                                        let buffer = buffer.clone();
+                                        let buffer = buffer.to_owned();
```
Comment (sdroege) [inline backends/gstreamer/audio_decoder.rs]: Why is this necessary, a whole buffer copy instead of just getting a new reference?

File: backends/gstreamer/render.rs [hunk 2]
```diff
@@ -96,9 +96,9 @@ impl GStreamerRender {
     }
 
     pub fn get_frame_from_sample(&self, sample: &gst::Sample) -> Result<Frame, ()> {
-        let buffer = sample.get_buffer().ok_or_else(|| ())?;
+        let buffer = sample.get_buffer().ok_or_else(|| ())?.to_owned();
```
Comment (sdroege) [inline backends/gstreamer/render.rs]: Same question here

**Comment by sdroege** [comment]
Generally it looks like the update improved the code a bit and made it nicer to read. Or what do you think? :)

File: backends/gstreamer/audio_decoder.rs (hunk 1 above)
Comment (sdroege) [reply backends/gstreamer/audio_decoder.rs]: There's `Sample::get_buffer_owned()` if that's the issue here, which allows to do this without a copy of the buffer. See https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/issues/204 for the background.

**Comment by ceyusa** [comment]
> Generally it looks like the update improved the code a bit and made it nicer to read. Or what do you think? :)

Agree, it feels more rusty, in the sense of the language, of course :)

Thanks @sdroege, specially for the `get_buffer_owned()` hint!

**Comment by Manishearth** [comment]
@bors-servo r+

Should also fix the unsafe code used in webrtcbin for https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/issues/189
